charms/kfp-api/src/charm.py

```diff
@@ -474,6 +501,14 @@ def _on_remove(self, _):
                 raise error
         self.unit.status = MaintenanceStatus("K8S resources removed")
 
+    def _on_update_status(self, _):
+        """Update status actions."""
+        self._check_leader()
```

Comment:
No way
That will put the app into error state

Reply:
Agreed with @beliaev-maksim, I think what you want is to put the `self._check_leader()` inside the try/except.  Right now `_check_leader()` will raise an `ErrorWithStatus` if we are not the leader.  That Exception is not caught here, so it goes to Juju and sets the unit in Error rather than in `WaitingStatus("Waiting for leadership")`

If this is in the try/except (or in its own try/except, which immediately returns after setting status in the exception), then it works as expected

Reply:
That's a typo/bug.